src/librbd/ImageCtx.h

```diff
@@ -68,6 +68,7 @@ namespace librbd {
                                         // a format librados can understand
     std::map<librados::snap_t, SnapInfo> snap_info;
     std::map<std::string, librados::snap_t> snap_ids;
+    std::map<cls::rbd::SnapshotNamespace, librados::snap_t> snap_namespace_ids;
```

Comment:
Please merge ```snap_ids ``` w/ ```snap_namespace_ids```. Use a std::pair of the namespace + the snapshot name for the unique key.